### ISSUE TITLE
Memory Leak with getText(“rawDICT”)

### ISSUE THREAD

**Issue by sshustov** [pymupdf/PyMuPDF#362]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
This bug was fixed in 1.14.15 version, but for some reason new fitz version has similar issue. This is same problem that appeared in the past issue: https://github.com/pymupdf/PyMuPDF/issues/290

## To Reproduce (mandatory)
I used same script from issue #290 

Tested 2 different versions: 1.16.1 (newest) and wheel PyMuPDF-1.14.15-cp27-cp27mu-manylinux1_x86_64.whl from this page https://github.com/pymupdf/PyMuPDF/releases/tag/1.14.15

```import fitz
import os
import psutil
import gc

file_path = 'any_pdf_document.pdf'
# Loop
niter = 300

memory_usage_start = [0] * niter
memory_usage_before_gc = [0] * niter
memory_usage_stop = [0] * niter

for i in range(niter):

    # Record initial memory usage
    process = psutil.Process(os.getpid())
    memory_usage_start[i] = process.memory_info().rss / 2 ** 20

    # Load file
    doc = fitz.open(file_path)
    page_count = doc.pageCount
    page = 0
    text = ''

    # Extract text
    while (page < page_count):
        p = doc.loadPage(page)
        page += 1
        # words = p.getTextWords()
        rawdict = p.getText('rawDict')
        # text += p.getText()

    # Record memory usage before cleanup
    memory_usage_before_gc[i] = process.memory_info().rss / 2 ** 20

    # Cleanup attempt
    # del text
    doc.close()
    gc.collect()

    # Record mem final usage
    memory_usage_stop[i] = process.memory_info().rss / 2 ** 20

# Viz mem usage VS iterations
from pylab import *
from matplotlib import pyplot as plt

iteration = list(range(0, niter))

plot(iteration[20:], memory_usage_start[20:])
plot(iteration[20:], memory_usage_before_gc[20:], 'r+')
plot(iteration[20:], memory_usage_stop[20:], 'g')

xlabel('iteration')
ylabel('memory usage (Mb)')
title('Memory usage over getText("rawDict")')
grid(True)

plt.savefig('memory_graph.png')
```

## Expected behavior (optional)
Graphs for getText() for both versions look similar. But when I used getText('rawDict') different versions gave different memory graph. It seems fix that was made on 1.14.15 does not exists in newest versions.

## Screenshots (optional)
Attached 4 screenshots

wheels 1.14.15 using getText()
![just_get_text_wheels1 14 15](https://user-images.githubusercontent.com/55201297/64720123-b39be780-d497-11e9-8bf7-01d4070f0b65.png)

1.16.1 using getText()
![just_get_text_version1 16 1](https://user-images.githubusercontent.com/55201297/64720122-b39be780-d497-11e9-8bcf-199fed773347.png)

wheels 1.14.15 using getText('rawDict')
![raw_dict_wheels1 14 15](https://user-images.githubusercontent.com/55201297/64720124-b39be780-d497-11e9-8883-576dc45e541e.png)

1.16.1 using getText('rawDict')
![version1 16 1](https://user-images.githubusercontent.com/55201297/64720125-b39be780-d497-11e9-8e0b-cad423f03330.png)

## Your configuration (mandatory)
 - Ubuntu 16.04 x64
 - python 2.7.12
 - PyMuPDF version (1.14.15 from wheel PyMuPDF-1.14.15-cp27-cp27mu-manylinux1_x86_64.whl), and 1.16.1 from pip install.

## Additional context (optional)
As I understand memory usage graphs should look similar for both versions, but it seems getText('rawDict') consumes much more memory than it should.
How long it would take to release new version with this bug fixed?

**Comment by JorjMcKie** [comment]
Weird - I was certain to not have changed that part.
Do you have the chance to test a Python 3 version? Would be gtreat.

There is a big difference between those versions: in Python 3 **all** memory handling is done by Python - even the internal one of MuPDF. Couldn't get this to work in Python 2 to date.

**Comment by sshustov** [comment]
I tested using python 3.6.8. Same situation. Wheel 1.14.15 has some fix for rawDict and I am not if it was added to new version.

![1 14 15 Wheels  Python 3 6 8  Memory usage over getText()](https://user-images.githubusercontent.com/55201297/64728409-e13d5c80-d4a8-11e9-923b-5eaf934e9ad7.png)
![1 16 1  Python 3 6 8  Memory usage over getText()](https://user-images.githubusercontent.com/55201297/64728414-e4d0e380-d4a8-11e9-90aa-8d481c1aa00d.png)
![1 14 15 Wheels  Python 3 6 8  Memory usage over getText( rawDict )](https://user-images.githubusercontent.com/55201297/64728422-e8646a80-d4a8-11e9-8341-ba1ab11ac2f2.png)
![1 16 1  Python 3 6 8  Memory usage over getText( rawDict )](https://user-images.githubusercontent.com/55201297/64728430-ea2e2e00-d4a8-11e9-8a26-47414e5ba6a6.png)
